stretch/rootfs/libversion.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+#
+# Library for managing versions strings
+
+# Functions
+########################
+# Gets version 
+# Arguments:
+#   $1 - string to extract major.minor.patch
+#   $2 - 1 to extract major, 2 to extract minor, 3 to extract patch
+# Returns:
+#   array with the major, minor and release
+#########################
+get_sematic_version () {
+    regex='([0-9]+)(\.([0-9]+)(\.([0-9]+))?)?'
+
+    if [[ $1 =~ $regex ]]; then
+        i=1
+        j=1
+        n=${#BASH_REMATCH[*]}
+
+        while [[ $i -lt $n ]]
+        do
+            if [ ! -z "${BASH_REMATCH[$i]}" ] && [ "${BASH_REMATCH[$i]:0:1}" != '.' ]
```

Comment:
```suggestion
            if [ -n "${BASH_REMATCH[$i]}" ] && [ "${BASH_REMATCH[$i]:0:1}" != '.' ]
```
You have negated [`test -z`](https://www.gnu.org/software/bash/manual/html_node/Bash-Conditional-Expressions.html#Bash-Conditional-Expressions), resulting in a needless double-negative